File: clang/test/SemaObjC/ns_returns_retained_block_return.m
// RUN: %clang_cc1  -fblocks -fsyntax-only -verify -Wno-objc-root-class %s
// RUN: %clang_cc1 -x objective-c++ -fblocks -fsyntax-only -verify -Wno-objc-root-class %s
// RUN: %clang_cc1  -fblocks -fobjc-arc -fsyntax-only -verify -Wno-objc-root-class %s
// RUN: %clang_cc1 -x objective-c++ -fblocks -fobjc-arc -fsyntax-only -verify -Wno-objc-root-class %s
// expected-no-diagnostics
// rdar://17259812

typedef void (^BT) ();

BT foo()  __attribute__((ns_returns_retained));

@interface I
BT foo()  __attribute__((ns_returns_retained));
@end

@implementation I
BT foo()  __attribute__((ns_returns_retained)) {return ^{}; }
@end

File: clang/test/SemaCXX/ns_returns_retained_block_return.cpp
// RUN: %clang_cc1 -fblocks -fsyntax-only -verify %s
// RUN: %clang_cc1 -fblocks -fobjc-arc -fsyntax-only -verify %s
// expected-no-diagnostics
// rdar://17259812

typedef void (^BT) ();

class S {
  BT br() __attribute__((ns_returns_retained)) {
    return ^{};
  }
 BT br1() __attribute__((ns_returns_retained));
};

BT S::br1() {
    return ^{};
}
